packages/dx-react-chart/src/plugins/zoom-pan.tsx

```diff
@@ -15,48 +12,66 @@ import {
 import { DragBox } from '../templates/drag-box';
 import {
   adjustLayout, getViewport, isKeyPressed, getOffset, getDeltaForTouches,
-  ScalesCache, getWheelDelta,
+  ScalesCache, getWheelDelta, getCoordsWithOffset,
 } from '@devexpress/dx-chart-core';
 import {
   ZoomAndPanProps, ZoomAndPanState, NumberArray, ZoomPanProviderProps, EventHandlers,
 } from '../types';
 
 const events = {
-  wheel: 'onWheel',
-  mousedown: 'onDown',
-  touchstart: 'onDown',
-  touchmove: 'onTouchMove',
-  touchend: 'onTouchEnd',
+  wheel: { func: 'onWheel' },
+  mousedown: {
+    func: 'onStart',
+    extraEvents: ['mousemove', 'mouseup'],
+  },
+  touchstart: {
+    func: 'onStart',
+    extraEvents: ['touchmove', 'touchend'],
+  },
 };
 
 class ZoomPanProvider extends React.PureComponent<ZoomPanProviderProps> {
-  ref!: Element;
   handlers!: EventHandlers;
+  ref!: Element;
+
   componentDidMount() {
     this.ref = this.props.rootRef.current!;
     this.handlers = Object.keys(events).reduce((prev, key) => {
+      const extraEvents = events[key].extraEvents;
       return {
         ...prev,
-        [key]: (e: any) => { this.props[events[key]](e); },
+        [key]: (e: any) => {
+          this.props[events[key].func](e);
+          if (extraEvents) {
+            const extraHandlers = {
+              [extraEvents[0]]: (event: any) => { this.props.onMove(event); },
+              [extraEvents[1]]: (event: any) => {
+                this.props.onEnd(event);
+                this.detachEvents(window, extraHandlers);
+              },
+            };
+            this.attachEvents(window, extraHandlers);
+          }
+        },
       };
     }, {});
-    this.attachEvents();
+    this.attachEvents(this.ref, this.handlers);
   }
 
-  attachEvents() {
-    Object.keys(this.handlers).forEach((el) => {
-      this.ref.addEventListener(el, this.handlers[el], { passive: false });
+  attachEvents(node, handlers) {
+    Object.keys(handlers).forEach((el) => {
+      node.addEventListener(el, handlers[el], { passive: false });
     });
   }
 
-  detachEvents() {
-    Object.keys(this.handlers).forEach((el) => {
-      this.ref.removeEventListener(el, this.handlers[el]);
+  detachEvents(node, handlers) {
+    Object.keys(handlers).forEach((el) => {
+      node.removeEventListener(el, handlers[el]);
     });
   }
 
   componentWillUnmount() {
```

Comment:
There may be events attached to `window` - they should be detached as well.
(It may happen when mouse is pressed and widget is destroyed before mouse is released)